src/table/cells/handle-key-press.js

```diff
@@ -67,48 +57,60 @@ export const preventDefaultBehavior = (evt) => {
   evt.preventDefault();
 };
 
-export const moveFocus = (evt, rootElement, rowIndex, colIndex, selState) => {
+export const moveFocus = (evt, rootElement, cellCoord, setFocusedCell, selState) => {
   preventDefaultBehavior(evt);
   removeCurrentFocus(evt);
   const rowAndColumnCount = getRowAndColumnCount(rootElement);
-  const { nextRow, nextCol } = arrowKeysNavigation(evt, rowAndColumnCount, rowIndex, colIndex, selState);
-  moveToNextFocus(rowAndColumnCount.rowElements, nextRow, nextCol);
+  const nextCellCoord = arrowKeysNavigation(evt, rowAndColumnCount, cellCoord, selState);
+  focusCell(rowAndColumnCount.rowElements, nextCellCoord);
+  setFocusedCell(nextCellCoord);
 };
 
-export const headHandleKeyPress = (evt, rootElement, rowIndex, colIndex, changeSortOrder, layout, isDim) => {
+export const headHandleKeyPress = (evt, rootElement, cellCoord, setFocusedCell, changeSortOrder, layout, isDim) => {
   switch (evt.key) {
     case 'ArrowUp':
     case 'ArrowDown':
     case 'ArrowRight':
     case 'ArrowLeft': {
-      !isCtrlShift(evt) && moveFocus(evt, rootElement, rowIndex, colIndex);
+      !isCtrlShift(evt) && moveFocus(evt, rootElement, cellCoord, setFocusedCell);
       break;
     }
     // Space bar / Enter: update the sorting
     case ' ':
     case 'Enter': {
       preventDefaultBehavior(evt);
-      changeSortOrder(layout, isDim, colIndex);
+      changeSortOrder(layout, isDim, cellCoord[1]);
       break;
     }
     default:
       break;
   }
 };
 
-export const bodyHandleKeyPress = (evt, rootElement, rowIndex, colIndex, cell, selState, selDispatch) => {
+export const bodyHandleKeyPress = (
+  evt,
+  rootElement,
+  cellCoord,
+  focusedCell,
+  setFocusedCell,
+  cell,
+  selState,
+  selDispatch
+) => {
   switch (evt.key) {
     case 'ArrowUp':
     case 'ArrowDown':
     case 'ArrowRight':
     case 'ArrowLeft': {
-      !isCtrlShift(evt) && moveFocus(evt, rootElement, rowIndex, colIndex, selState);
+      !isCtrlShift(evt) && moveFocus(evt, rootElement, cellCoord, setFocusedCell, selState);
       break;
     }
     // Space bar: Selects value.
     case ' ': {
       preventDefaultBehavior(evt);
-      cell?.isDim && selectCell(cell, selState, selDispatch, evt);
+      cell?.isDim &&
+        handleCellFocus(cell, focusedCell, setFocusedCell, rootElement) &&
```

Comment:
Does it need to be called here?

Reply:
clicking is the same as the space key, right?

Reply:
well, almost. Clicking can both move focus and make selection. Pressing space should only select, since you already have navigated to that cell. The focus state should be correct before pressing space

Reply:
The focus state is not correct before pressing space. Pressing space should also resetTabIndex.